certification/internal/shell/imagesource_registry.go

```diff
@@ -0,0 +1,81 @@
+package shell
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	cmdchain "github.com/rainu/go-command-chain"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	log "github.com/sirupsen/logrus"
+)
+var userRegistry string
+
+type ImageSourceRegistryCheck struct {
+}
+
+var approvedRegistries = map[string]string{
+	"registry.connect.dev.redhat.com":   "registry.connect.dev.redhat.com",
+	"registry.connect.qa.redhat.com":    "registry.connect.qa.redhat.com",
+	"registry.connect.stage.redhat.com": "registry.connect.stage.redhat.com",
+	"registry.connect.redhat.com":       "registry.connect.redhat.com",
+	"registry.redhat.io":                "registry.redhat.io",
+	"registry.access.redhat.com":        "registry.access.redhat.com",
+}
+
+func (p *ImageSourceRegistryCheck) Validate(bundleImage string) (bool, error) {
+
+	output := &bytes.Buffer{}
+	inputContent := strings.NewReader(bundleImage)
+
+	err := cmdchain.Builder().
+		WithInput(inputContent).
+		Join("cut", "-d", ",", "-f1").
+		Join("cut", "-d", "/", "-f1").
+		Finalize().WithOutput(output).Run()
+	if err != nil {
+		log.Error(" Failed to execute cmdchain builder")
+		log.Debug(" failed to execute cmdchain builder", err)
+		return false, nil
+	}
+
+	userRegistry = strings.TrimRight(output.String(), "\n")
+	log.Info("Check Image registry for : ", userRegistry)
+
+	if val, ok := approvedRegistries[userRegistry]; ok {
+		log.Debug(userRegistry, "Found "+val+" in approved registry")
```

Comment:
IMO, using fmt.Sprintf will improve the code readability for the second arg.

Reply:
Agreed and it's done.